backend/src/app.ts

```diff
@@ -119,4 +120,27 @@ app.get('/reviews', async (req, res) => {
   }
 });
 
+app.get('/homepageData', async (req, res) => {
+  const buildingDocs = (await aptCollection.limit(3).get()).docs;
+  const buildings: Apartment[] = buildingDocs.map((doc) => doc.data() as Apartment);
+  // eslint-disable-next-line consistent-return
```

Comment:
these are a lot of linter disables. i know that you want to do things in one-pass (so filtering and then mapping wouldn't be ideal), but you can still do it in one-pass without having these linter disables using `reduce` - try accumulating from an empty array and pushing to it when the condition is met